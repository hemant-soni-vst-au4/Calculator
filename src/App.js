import React from 'react';

const App = () => (
  <div>
    <h1>Calculator Application</h1>
  </div>
);

export default App;
