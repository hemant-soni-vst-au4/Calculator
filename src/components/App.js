import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate'; // eslint-disable-line

const App = () => (
  <>
    <Display />
    <ButtonPanel />
  </>
);

export default App;
