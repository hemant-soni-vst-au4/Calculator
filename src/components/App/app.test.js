import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import App from './App';

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('App', () => {
  test('renders App component', () => {
    render(<App />);
    expect(screen.getByText('AC')).toBeInTheDocument();
  });

  test('renders Div correctly', () => {
    const tree = renderer.create(<div className="app" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
