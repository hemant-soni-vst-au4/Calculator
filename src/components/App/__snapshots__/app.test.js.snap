// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="display"
  >
    0
  </div>
  <div
    className="button-panel"
  >
    <div>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        className=" orange"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        รท
      </button>
    </div>
    <div>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        className=" orange"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        x
      </button>
    </div>
    <div>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        className=" orange"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        className=" orange"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div>
      <button
        className="wide lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        className=" lightgray"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "lightgray",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        className=" orange"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
